controllers/front/action.php

```diff
@@ -17,25 +18,366 @@
+                $wishlist->id_customer = $this->context->customer->id;
+                $wishlist->name = 'default';
+                $wishlist->token = $this->generateWishlistToken();
+                $wishlist->default = 1;
+                $wishlist->add();
+            }
+        } else {
+            $wishlist = new Wishlist($idWishlist);
+        }
+
+        $productIsAdded = $wishlist->addProduct(
+            $idWishlist,
+            $this->context->customer->id,
+            $id_product,
+            $id_product_attribute,
+            $quantity
+        );
+
+        if ($productIsAdded === false) {
+            $this->ajaxDie(
+                Tools::jsonEncode([
+                    'success' => false,
+                    'message' => $this->module->l('There was an error adding the product', 'mywishlist'),
+                ])
+            );
+        }
+
+        $this->ajaxDie(
+            Tools::jsonEncode([
+                'success' => false,
+                'message' => $this->module->l('Product added', 'mywishlist'),
+            ])
+        );
+    }
+
+    public function createNewWishlistAction($params)
+    {
+        if (isset($params['name'])) {
+            $wishlist = new Wishlist();
+            $wishlist->name= $params['name'];
+            $wishlist->id_shop_group = $this->context->shop->id_shop_group;
+            $wishlist->id_customer = $this->context->customer->id;
+            $wishlist->id_shop = $this->context->shop->id;
+            $wishlist->token = $this->generateWishlistToken();
+
+            if (true === $wishlist->save()) {
+                $this->ajaxDie(
+                    Tools::jsonEncode([
+                        'status' => 'success',
+                        'message' => $this->module->l('The list has been properly created', 'mywishlist'),
+                        'datas' => [
+                            'name' => $wishlist->name,
+                            'id' => $wishlist->id
+                        ]
+                    ])
+                );
+            }
+
+            $this->ajaxDie(
+                Tools::jsonEncode([
+                    'success' => false,
+                    'message' => $this->module->l('Error saving the new wishlist', 'mywishlist')
+                ])
+            );
+        } else {
+            $this->ajaxDie(
+                Tools::jsonEncode([
+                    'success' => false,
+                    'message' => $this->module->l('Missing name parameter', 'mywishlist')
+                ])
+            );
+        }
+    }
+
+    public function renameWishlistAction($params)
+    {
+        if (isset($params['idWishlist'], $params['name'])) {
+            $wishlist = new wishlist($params['idWishlist']);
+            $wishlist->name = $$params['name'];
+
+            if (true === $wishlist->save()) {
+                $this->ajaxDie(
+                    Tools::jsonEncode([
```

Comment:
```suggestion
                    json_encode([
```
https://github.com/PrestaShop/PrestaShop/blob/1.7.6.x/classes/Tools.php#L3007